packages/navie/src/lib/mermaid-filter.ts

```diff
@@ -0,0 +1,116 @@
+import assert from 'assert';
+
+import Filter, { Chunk } from './filter';
+import MermaidValidator from './mermaid-validator';
+
+import InteractionHistory from '../interaction-history';
+import MermaidFixerService from '../services/mermaid-fixer-service';
+
+export default class MermaidFilter implements Filter {
+  private diagram: string[] | undefined;
+  private buffer = '';
+
+  constructor(
+    private readonly history: InteractionHistory,
+    private readonly mermaidFixer: MermaidFixerService
+  ) {}
+
+  get inDiagram() {
+    return this.diagram !== undefined;
+  }
+
+  async *transform(chunk: string): AsyncIterable<Chunk> {
+    // Lines are delineated by '\n'.
+    // When a new chunk is received, append the chunk to the buffer.
+    // Extract any complete lines from the buffer and process them.
+
+    this.buffer += chunk;
+
+    while (this.buffer.indexOf('\n') !== -1) {
+      const nextLine = this.buffer.slice(0, this.buffer.indexOf('\n'));
+      this.buffer = this.buffer.slice(this.buffer.indexOf('\n') + 1);
+      yield* this.processLine(nextLine);
+    }
```

Comment:
`nextLine` is off by one, dropping the new line.

```suggestion
    for (;;) {
      const lineEndIndex = this.buffer.indexOf('\n');
      if (lineEndIndex === -1) break;

      const nextLine = this.buffer.slice(0, lineEndIndex + 1);
      this.buffer = this.buffer.slice(lineEndIndex + 1);
      yield* this.processLine(nextLine);
    }
```

Reply:
It seems to be adding extra newlines: https://github.com/getappmap/appmap-js/actions/runs/9963827482/job/27530751071#step:5:2601

Reply:
@dustinbyrne is there still a code change required here or did it turn out to be ok as written?